dynamic_sampling_context.go

```diff
@@ -36,10 +37,57 @@ func NewDynamicSamplingContext(header []byte) (DynamicSamplingContext, error) {
 	}, nil
 }
 
+func DynamicSamplingContextFromTransaction(span *Span) (DynamicSamplingContext, error) {
+	entries := map[string]string{}
+
+	hub := hubFromContext(span.Context())
+	scope := hub.Scope()
+	options := hub.Client().Options()
+
+	if traceID := span.TraceID.String(); traceID != "" {
+		entries["trace_id"] = traceID
+	}
+	if sampleRate := span.sampleRate; sampleRate != 0 {
+		entries["sample_rate"] = strconv.FormatFloat(sampleRate, 'f', -1, 64)
+	}
+
+	if dsn := hub.Client().dsn; dsn != nil {
+		if publicKey := dsn.publicKey; publicKey != "" {
+			entries["public_key"] = publicKey
+		}
+	}
+	if release := options.Release; release != "" {
+		entries["release"] = release
+	}
+	if environment := options.Environment; environment != "" {
+		entries["environment"] = environment
+	}
+
+	// Only include the transaction name if it's of good quality (not empty and not SourceURL)
+	if span.Source != "" && span.Source != SourceURL {
+		if transactioName := scope.Transaction(); transactioName != "" {
+			entries["transaction"] = transactioName
+		}
+	}
+
+	if userSegment := scope.user.Segment; userSegment != "" {
+		entries["user_segment"] = userSegment
```

Comment:
m: should we gate this by `sendDefaultPII`?

Reply:
Not sure, did we do this in other SDKs?
In the end, this is explicitly set by the user.

Reply:
You're right, let's keep this then.